components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManagerImpl.java

```diff
@@ -94,6 +94,31 @@ public void createFederatedAssociationWithIdpResourceId(User user, String idpId,
         createFederatedAssociation(user, idpName, federatedUserId);
     }
 
+    @Override
+    public User getAssociatedLocalUser(String tenantDomain, String idpId, String federatedUserId)
+            throws FederatedAssociationManagerException {
+
+        // Resolve idp name from IDP UUID.
+        String idpName = getIdentityProviderNameByResourceId(idpId);
+        if (StringUtils.isEmpty(idpName)) {
+            throw handleFederatedAssociationManagerClientException(INVALID_IDP_PROVIDED, null, true);
+        }
+        String usernameWithDomain = getUserForFederatedAssociation(tenantDomain, idpName, federatedUserId);
+        if (StringUtils.isNotBlank(usernameWithDomain)) {
+            User user = new User();
+            if (usernameWithDomain.contains(UserCoreConstants.DOMAIN_SEPARATOR)) {
```

Comment:
this should be redundant. please check

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/3586/commits/d4f392a98ee661912256bce2ab9a93720df6a5aa